### ISSUE TITLE
[BUG] keycloak configuration just gives a black screen, no dashboard, no login prompt

### ISSUE THREAD

**Issue by lgwapnitsky** [Lissy93/dashy#520]
### Environment

Self-Hosted (Bare Metal)

### Describe the problem

I've configured dashy to authenticate against my keycloak instance, which is currently working properly for other apps.

Upon rebuilding and relaunching dashy, I get a black screen, no login prompt, nothing else. The logs do not show anything, and a config validation shows no errors

Se

### Additional info

```
appConfig:
  theme: one-dark
  layout: vertical
  iconSize: medium
  language: en
  auth:
    enableKeycloak: true
    keycloak:
      serverUrl: https://auth.wapnet.MYDOMAIN.com
      realm: 'master'
      clientId: 'dashy'
```
![Screenshot at 2022-02-28 22-19-24](https://user-images.githubusercontent.com/647444/156098551-935094ea-d8f2-49c8-b71e-823610e14114.png)

### Please tick the boxes

- [X] You are using a [supported](https://github.com/Lissy93/dashy/blob/master/.github/SECURITY.md#supported-versions) version of Dashy (check the first two digits of the version number)
- [X] You've checked that this [issue hasn't already been raised](https://github.com/Lissy93/dashy/issues?q=is%3Aissue)
- [X] You've checked the [docs](https://github.com/Lissy93/dashy/tree/master/docs#readme) and [troubleshooting](https://github.com/Lissy93/dashy/blob/master/docs/troubleshooting.md#troubleshooting) guide
- [X] You agree to the [code of conduct](https://github.com/Lissy93/dashy/blob/master/.github/CODE_OF_CONDUCT.md#contributor-covenant-code-of-conduct)

**Comment by lgwapnitsky** [comment]
I"ve followed all steps, no luck, Please advise

**Comment by Lissy93** [comment]
In future, please provide the log output in the Additional Info section. It's not possible to diagnose an issue without an error output.

But I don't think this is a bug, it sound like a CORS error (see [troubleshooting](https://github.com/Lissy93/dashy/blob/master/docs/troubleshooting.md#keycloak-redirect-error) docs), which can be fixed by setting the correct headers to your Keycloak instance. It will likely be `Access-Control-Allow-Origin [URL-of Dashy]`. For a tutorial on how to do so, see [Setting Headers](https://github.com/Lissy93/dashy/blob/master/docs/management.md#setting-headers) in the Management docs, or check the docs for your proxy / server.

In future, please also take a look for similar issues. This looks similar to: #409, #479, #491, #507, #341
There's also a lot of details in the documentation which you should check out (because it took me ages to write!)

Hope that helps, I'm going to keep this ticket closed, as nothing to be fixed on Dashy's side.

**Comment by lgwapnitsky** [comment]
I believe I see part of the issue. I'm running the latest version of
keycloak, which does not append "auth" to the beginning of the URL!

**Comment by lgwapnitsky** [comment]
Adjusted the keycloak js file, and am making small progress. Now
encountering issue with frame-ancestor violations.

Here is a copy of my HAPROXY backend config for my dashboard:

# Backend: DASHY_backend ()
backend DASHY_backend
    # health checking is DISABLED
    mode http
    balance source
    # stickiness
    stick-table type ip size 50k expire 30m
    stick on src
    # tuning options
    timeout connect 30s
    timeout server 30s
    # ACL: DASHY_cond
    acl acl_621e323067bda5.67311074 hdr(host) -i dashboard.wapnitsky.com

    # ACTION: DASHY_Header_rule
    http-request add-header Access-Control-Allow-Origin
https://dashboard.wapnitsky.com if acl_621e323067bda5.67311074
    # ACTION: DASHY_frame_ancestors_rule
    http-request add-header Content-Security-Policy "frame-ancestors
auth.wapnet.wapnitsky.com dashboard.wapnitsky.com" if
acl_621e323067bda5.67311074
    http-reuse safe
    server dashy 10.150.88.51:4000



and here is a screenshot of my keycloak config:

[image: image.png]

I appreciate your hard work in this project. I love it, and am looking
forward to getting keycloak 17 to work with it :)

**Comment by Lissy93** [comment]
What error do you see in [the console](https://github.com/Lissy93/dashy/blob/master/docs/troubleshooting.md#how-to-open-browser-console)?

**Comment by lgwapnitsky** [comment]
[image: image.png]

*Larry G. Wapnitsky*


*E: ***@***.****
*Web: Larry.Wapnitsky.com <http://larry.wapnitsky.com/>Twitter:
@LGWapnitsky*

**Comment by lgwapnitsky** [comment]
I believe it's due to the npm file. There is a new one released a few weeks
ago that appears to be written with the new version in mind.

I may make a pull request when I can find some spare time and see if i can
rewrite for 17 using the new module

**Comment by lgwapnitsky** [comment]
Progress made using 'http-relative-path=/auth' in my config file. Now
dealing with a keycloak page not being found at the moment for
/auth/realms/master/protocol/openid-connect

**Comment by lgwapnitsky** [comment]
All good now.

**Comment by Lissy93** [comment]
Glad you've got it working :)
Would you be up for adding your solution to the troubleshooting guide, to help others who might face the same issue? Or explain it here, and I can add it in.

**Comment by lgwapnitsky** [comment]
Will do. Give me a.few days.

**Comment by lgwapnitsky** [comment]
OK. In order to get it to work, I had to expose the auth relative path in
the keycloak 17 configuration

http-relative-path=/auth

Unfortunately, this breaks the keycloak user account console. Once I remove
that, auth with dashy stops working, but the console starts.

Is it possible to test with the newer npms that were released in February
2022 that support KC17?

keycloak-connect          | Keycloak Connect_    | =keycloak.bot_  |
2022-02-10 | 17.0.0   | sso keycloak oauth oauth2.0 authentication
keycloak-js               | Keycloak Adapter     | =keycloak.bot_  |
2022-02-12 | 17.0.0   | keycloak sso oauth oauth2 authentication
